src/plugins/data/public/ui/query_editor/editors/default_editor/index.tsx

```diff
@@ -5,7 +5,7 @@
 
 import React from 'react';
 import { EuiText, EuiTextColorProps } from '@elastic/eui';
-// import { monaco } from '@osd/monaco';
+import { monaco } from '@osd/monaco';
```

Comment:
how come we did this?

@osd/monaco is the wrapper around the monaco editor. and the CodeEditor is the component that wires up the monaco editor. and then everyone should interact with the CodeEditor component. This way, if someone in another plugin wants to use the autocomplete and the code editor but don't want to use the data plugin UI then they can still have all that ready to go. but this would require developers to do the same code we have here.

we should refactor to ensure the CodeEditor is the source of truth.

Reply:
Not sure if I understand the suggestion. The monaco import here is only for specifying the correct type. If you take a look at few lines below
```
provideCompletionItems: monaco.languages.CompletionItemProvider['provideCompletionItems'];
``` 
that this is the only usage. The autocomplete itself is only hooked in on CodeEditor. What would be the suggestion if this is not what you are looking at?